aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -251,6 +255,154 @@ void gemm(
       transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
 
+template <typename scalar_t>
+static void gemm_batched_mkl_impl(
+      TransposeType transa, TransposeType transb,
+      int64_t batch_size, int64_t m, int64_t n, int64_t k,
+      scalar_t alpha,
+      const scalar_t **a, int64_t lda,
+      const scalar_t **b, int64_t ldb,
+      scalar_t beta,
+      scalar_t **c, int64_t ldc) {
+  for (int64_t i = 0; i < batch_size;) {
+    int sub_batch = std::min(batch_size - i, int64_t{INT_MAX});
+    mkl_gemm_batched(transa, transb, sub_batch, m, n, k, alpha,
+                     &a[i], lda, &b[i], ldb, beta, &c[i], ldc);
+    i += sub_batch;
+  }
+}
+
+template <typename scalar_t>
+using is_blas_library_type = std::integral_constant<bool,
+    std::is_same<scalar_t, double>::value ||
+    std::is_same<scalar_t, float>::value ||
+    std::is_same<scalar_t, c10::complex<double>>::value ||
+    std::is_same<scalar_t, c10::complex<float>>::value>;
+
+template <typename scalar_t>
+void gemm_batched_generic(
+    TransposeType transa, TransposeType transb,
+    int64_t batch_size, int64_t m, int64_t n, int64_t k,
+    scalar_t alpha,
+    const scalar_t **a, int64_t lda,
+    const scalar_t **b, int64_t ldb,
+    scalar_t beta,
+    scalar_t **c, int64_t ldc) {
+  for (int64_t batch = 0; batch < batch_size; ++batch) {
+    gemm(transa, transb, m, n, k, alpha, a[batch], lda, b[batch], ldb, beta, c[batch], ldc);
+  }
+}
+
+template <typename scalar_t>
+void gemm_batched(
+    TransposeType transa, TransposeType transb,
+    int64_t batch_size, int64_t m, int64_t n, int64_t k,
+    scalar_t alpha,
+    const scalar_t **a, int64_t lda,
+    const scalar_t **b, int64_t ldb,
+    scalar_t beta,
+    scalar_t **c, int64_t ldc) {
+  if (batch_size == 1) {
+    return gemm(transa, transb, m, n, k, alpha, a[0], lda, b[0], ldb, beta, c[0], ldc);
```

Comment:
this is nice! I was wondering if calling gemm_batched regardles of `groups` is going to be ok, but with this special case it is